src/pytest_reportlog/plugin.py

```diff
@@ -67,6 +67,16 @@ def pytest_runtest_logreport(self, report):
         )
         self._write_json_data(data)
 
+    def pytest_warning_recorded(self, warning_message, when, nodeid, location):
+        data = vars(warning_message)
+        extra_data = {
+            "$report_type": "warning-recorded",
```

Comment:
> what should the value for $report_type be? In https://github.com/pytest-dev/pytest-reportlog/issues/20, it was suggested to use warning-recorded, but I wonder if WarningRecorded would be closer to the values of other report types?

Agreed!

```suggestion
            "$report_type": "WarningMessage",
```

Lets use "WarningMessage" as thats the name of the class.

Reply:
no strong opinion here, but for the other `$report_type`s the entire `dict` is a serialization of the class, while for `WarningMessage` we'd have additional data (`when`, `location`), and depending on whether we include all information or not, we'd also be missing a few fields. So really, to be consistent we'd have to add a new class to `pytest` and dump that here.

Reply:
You got a good point, but I would say we never made any promises that the dict contents were guaranteed to map directly to the class attributes (giving the impression that one can recreate the class using the dictionary), the contents are more derived from  the classes than anything else -- it just so happens that so far we did a dump of all the attributes.

Even if we do change that in the future, I would prefer to keep the file format consistent, even if we change things behind the scenes later.

Reply:
makes sense, thanks.